src/titlecase/full.rs

```diff
@@ -0,0 +1,416 @@
+                .as_bytes()
+                .as_bstr()
+        );
+
+        // Change length when titlecased
+        // https://github.com/minimaxir/big-list-of-naughty-strings/blob/894882e7/blns.txt#L226-L232
+        let s = "ZȺȾ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(
+            iter.collect::<Vec<_>>().as_bstr(),
+            "Zⱥⱦ".as_bytes().as_bstr()
+        );
+    }
+
+    #[test]
+    fn invalid_utf8() {
+        let iter = Titlecase::from(&b"\xFF\xFE"[..]);
+        assert_eq!(iter.collect::<Vec<u8>>().as_bstr(), b"\xFF\xFE".as_bstr());
+
+        let iter = Titlecase::from(&b"ABC\xFF\xFEXYZ"[..]);
+        assert_eq!(
+            iter.collect::<Vec<u8>>().as_bstr(),
+            b"Abc\xFF\xFExyz".as_bstr()
+        );
+
+        let iter = Titlecase::from(&b"abc\xFF\xFEXYZ"[..]);
+        assert_eq!(
+            iter.collect::<Vec<u8>>().as_bstr(),
+            b"Abc\xFF\xFExyz".as_bstr()
+        );
+
+        // The bytes \xF0\x9F\x87 could lead to a valid UTF-8 sequence, but 3 of
+        // them on their own are invalid. Only one replacement codepoint is
+        // substituted, which demonstrates the "substitution of maximal
+        // subparts" strategy.
+        //
+        // See: https://docs.rs/bstr/1.*/bstr/#handling-of-invalid-utf-8
+        let iter = Titlecase::from(&b"aB\xF0\x9F\x87Yz"[..]);
+        assert_eq!(
+            iter.collect::<Vec<_>>().as_bstr(),
+            b"Ab\xF0\x9F\x87yz".as_bstr()
+        );
+    }
+
+    #[test]
+    fn unicode_replacement_character() {
+        let s = "�".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(iter.collect::<Vec<_>>().as_bstr(), "�".as_bytes().as_bstr());
+    }
+
+    #[test]
+    fn dz_titlecase() {
+        let s = "ǅ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(iter.collect::<Vec<_>>().as_bstr(), "ǅ".as_bytes().as_bstr());
+    }
+
+    #[test]
+    fn latin_capital_i_with_dot_above() {
+        let s = "İ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(iter.collect::<Vec<_>>().as_bstr(), "İ");
+    }
+
+    #[test]
+    fn case_map_to_two_chars() {
+        let s = "և".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(
+            iter.collect::<Vec<_>>().as_bstr(),
+            "Եւ".as_bytes().as_bstr()
+        );
+
+        let s = "ẙ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(
+            iter.collect::<Vec<_>>().as_bstr(),
+            "Y\u{30a}".as_bytes().as_bstr()
+        );
+
+        let s = "ᾂ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(iter.collect::<Vec<_>>().as_bstr(), "ᾊ".as_bytes().as_bstr());
+
+        let s = "ﬗ".as_bytes();
+        let iter = Titlecase::from(s);
+        assert_eq!(
+            iter.collect::<Vec<_>>().as_bstr(),
+            "Մխ".as_bytes().as_bstr()
+        );
+    }
+
+    #[test]
+    fn case_map_to_three_chars() {
```

Comment:
IMO, I would move the test cases that don't look right to the correct test case. e.g. moving:

```
ᾂ -> ἊΙ / ᾊ [8066] -> [7946, 921] / [8074]
```

to the `case_map_to_two_chars` based on what you've got here.

I suspect there might be some unicode [de]normalization going on here since U+1F82 can be denormalized into 4 codepoints: https://www.compart.com/en/unicode/U+1F82, and thus I suspect there is some other combinations around for this.

Reply:
> I would move the test cases that don't look right to the correct test case.

Ah, that make sense. Fixed in 09e6fe5df25cdd0d37b37bf7c8396d7a9487bcbf.

> I suspect there might be some unicode [de]normalization going on here since U+1F82 can be denormalized into 4 codepoints: https://www.compart.com/en/unicode/U+1F82, and thus I suspect there is some other combinations around for this.

I'm not sure if Unicode composition is related to case mapping 🤔 IIUC titlecasing is just a simple one directional mapping similar to uppercasing and lowercasing. 

https://www.compart.com/en/unicode/U+1F82 only shows its `Uppercase Character`, but https://codepoints.net/U+1F82 does show all its three kinds of case mapping.